### ISSUE TITLE
word language model never varies text produced.

### ISSUE THREAD

**Issue by wesboyt** [pytorch/examples#829]
No matter which random input token generate.py is handed, the same generated.txt file is produced. 
at 
https://github.com/pytorch/examples/blob/master/word_language_model/generate.py#L56

I replaced input with each onehot in range and produced the same final paragraph no matter what.
"input = torch.Tensor([[789]]).long().to(device)"

**Comment by YoelShoshan** [comment]
I just trained now a basic transformer model using this example code, and I don't have the issue that you describe. 
For example, for the same rng seed:

When using:
input = torch.Tensor([[123]]).long().to(device)
and when using
input = torch.Tensor([[12300]]).long().to(device)
I get completely different generated text.

I'm far from being NLP expert, but I have a suggestion - is it possible that your model is under-trained? 
what are your exact train args ?

This are the exact args I used for training:
python main.py --cuda --model Transformer --lr 5 --emsize 1500 --epochs 40
